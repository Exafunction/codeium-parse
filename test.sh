#!/bin/bash
set -euo pipefail

cd "$(dirname "${BASH_SOURCE[0]}")"
for test_file in test_files/*; do
    test_file="$(basename "$test_file")"
    echo "$test_file"
    ./parse -file "test_files/$test_file" -use_tags_query -tags_query_dir "queries" > "goldens/$test_file.golden.tmp"
    trap 'rm "goldens/$test_file.golden.tmp"' EXIT
    diff -u "goldens/$test_file.golden" "goldens/$test_file.golden.tmp"
done
