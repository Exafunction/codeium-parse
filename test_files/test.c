#include "runtime/parser.h"

typedef struct {
  Parser* parser;
  TSSymbol lookahead_symbol;
} SkipPrecedingTreesSession;

static void parser__push(Parser* self, StackVersion version, Tree* tree,
                         TSStateId state) {
  ts_stack_push(self->stack, version, tree, false, state);
  ts_tree_release(tree);
}

static Tree* parser__lex(Parser* self, StackVersion version) {
  TSStateId parse_state = ts_stack_top_state(self->stack, version);
  Length start_position = ts_stack_top_position(self->stack, version);
  TSLexMode lex_mode = self->language->lex_modes[parse_state];
}
