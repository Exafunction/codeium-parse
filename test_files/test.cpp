#include <string>

namespace bar {
// Declaration.
bool baz(const std::string& bing);

struct Foo2;

}  // namespace bar

namespace bar::bar2 {
struct Foo3;
}

class Foo {
 public:
  // foo
  // foo2
  void foo() {
    // Do nothing.
  }
};

struct bar::Foo2 {};
struct bar::bar2::Foo3 {};

// baz
bool bar::baz(const std::string& bing) {
  // This shouldn't be recognized as a function.
  Foo dist();
  return bing.empty();
}

// one_pointer comment
bool* one_pointer() { return nullptr; }
// two_pointer comment
bool** two_pointer() { return nullptr; }

// Comment.
template <typename T>
class TemplatedFoo {
  template <typename U>
  void foo();
};

// Comment.
template <typename T>
template <typename U>
void TemplatedFoo<T>::foo() {}

// Comment.
template <>
template <>
void TemplatedFoo<int>::foo<int>() {}

// Comment.
template <typename T>
class TemplatedFoo2;

// Comment.
template <>
class TemplatedFoo2<int> {};
