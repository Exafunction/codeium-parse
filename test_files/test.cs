// clang-format off
namespace Namespace {
  // Comment A.
  internal delegate void A(params int[] test);
  // Comment F.
  public struct F<T>
      where T : struct {
  }
  // Comment F.
  [Nice]
  private record F<T1, T2>
      where T1 : I1, I2, new()
      where T2 : I2 {}
  // Comment Teacher.
  record Teacher(string FirstName, string LastName, string Subject) : Person(FirstName, LastName);
  // Comment B.
  enum B { Ten = 10, Twenty = 20 }
  // Comment F.
  public class F : object, IAlpha, IOmega {}
  // Comment Class.
  public partial class Class<in TParam>
      where TParam : class ?, notnull, F ? {
    // Comment operator +.
    public static int operator +(A a) {
      return 0;
    }
    // Comment GetSet.
    uint GetSet { get; set; }
    // Comment Foo.
    static extern Foo() => bar = 0;
    // Comment ~Class.
    extern ~Class() {}
    // Comment Bar
    public void Bar() => bar = 0;
  }
  // Comment IFoo.
  public interface IFoo {}
}
// clang-format on
