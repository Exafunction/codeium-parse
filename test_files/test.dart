import 'package:test/test.dart';

// Functions

// foo not a docstring
void foo() {
    print("Hello, World!");
}

/// bar is a docstring
int bar() {
    return 0;
}

// Classes with methods

class FooClass {
    int someField = 0;

    // Constructor
    FooClass() {
        this.someField = 0;
    }

    /// FCB docstring
    int fooClassBar() {
        return 0;
    }
}

// Mixins
mixin FooMixin {
    int barMixin() {
        return 0;
    }
}

// Combinations
class FooBar extends FooClass with Foo {
    int bar() {
        return 0;
    }
}

void main() {
    print("Hello, World!");
}
