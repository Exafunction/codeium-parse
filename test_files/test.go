package test_files

// Pattern 0.
type Pattern0 struct {
	a string
}

// Pattern 1.
type Pattern1 interface {
	a() string
}

// Unsupported.
type (
	UnsupportedStruct    struct{}
	UnsupportedInterface interface{}
)

func Pattern2() {}

// Comments
// here.
func (p *Pattern0) Method(string) {}

func (p Pattern0) Method2(a string) string {
	Pattern2()
	s := &Pattern0{}
	s.Method("")
	((*Pattern0)(s)).Method("")
	var s2 any
	s2.(*Pattern0).Method("")
	return a
}

type Alias string
