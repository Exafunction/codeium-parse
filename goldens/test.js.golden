Name: "path/to/file/1"
Definition (definition.import):
import { named1, named2 as alias2, named3 as alias3 } from "path/to/file/1";
Named imports:
    named1
    named2 as alias2
    named3 as alias3

Name: "path/to/file/2"
Definition (definition.import):
import default1 from "path/to/file/2";
Default import: default1

Name: 'npm-lib-1'
Definition (definition.import):
import default2, { named4 as alias4 } from 'npm-lib-1'
Default import: default2
Named imports:
    named4 as alias4

Name: "npm-lib-2"
Definition (definition.import):
import * as entireModule from "npm-lib-2";
Namespace: entireModule

Name: "./side-effects-only-1.js"
Definition (definition.import):
import "./side-effects-only-1.js";

Name: Pattern1a
Doc:
/**
 * Pattern 1a comment.
 */
Definition (definition.class):
class Pattern1a {
  constructor() {}
  /**
   * Pattern 0 comment.
   */
  pattern0(param1) {
    const moment = require('moment');
    return param1;
  }
}

Name: pattern0
Parameters: (param1)
Doc:
/**
   * Pattern 0 comment.
   */
Definition (definition.method):
pattern0(param1) {
    const moment = require('moment');
    return param1;
  }
Lineage: [Pattern1a]

Name: Pattern1b
Doc:
// Pattern 1b comment.
Definition (definition.class):
export class Pattern1b {}
Metadata:
    is_export = true

Name: pattern2
Parameters: (param1)
Doc:
// Pattern 2 comment.
// More pattern 2 comment.
/* Even more
 * pattern 2 comment.
 */
Definition (definition.function):
function pattern2(param1) {
  return param1;
}

Name: f
Parameters: ()
Doc:
// f comment.
Definition (definition.function):
export function f() {}
Metadata:
    is_export = true

Name: g
Parameters: ()
Doc:
// g comment.
Definition (definition.function):
function* g() {}

Name: pattern3
Parameters: (param1)
Doc:
/**
 * Pattern 3 comment.
 */
Definition (definition.function):
export default function* pattern3(param1) {
  yield param1;
}
Metadata:
    is_export = true

Name: pattern4a
Parameters: (param1)
Doc:
/**
 * Pattern 4a comment.
 */
Definition (definition.function):
const pattern4a = function (param1) {
  return param1;
};

Name: pattern4b
Parameters: (param1)
Doc:
/**
 * Pattern 4b comment.
 */
Definition (definition.function):
export const pattern4b = (param1) => {};
Metadata:
    is_export = true

Name: pattern5a
Parameters: (param1)
Doc:
/**
 * Pattern 5a comment.
 */
Definition (definition.function):
export var pattern5a = function (param1) {
  return param1;
};
Metadata:
    is_export = true

Name: pattern5b
Parameters: (param1)
Doc:
/**
 * Pattern 5b comment.
 */
Definition (definition.function):
var pattern5b = (param1) => {};

Name: pattern6a
Parameters: (a)
Doc:
// Pattern 6a comment.
Definition (definition.function):
pattern6a = function(a) {};

Name: pattern6b
Parameters: (b)
Doc:
// Pattern 6b comment.
Definition (definition.function):
pattern6b = (b) => {};

Name: member
Parameters: (c)
Doc:
// Pattern 6c comment.
Definition (definition.function):
pattern6c.member = function(c) {};

Name: member
Parameters: (d)
Doc:
// Pattern 6d comment.
Definition (definition.function):
pattern6d.member = (d) => {};

Name: pattern7a
Parameters: (a)
Doc:
// Pattern 7a comment.
Definition (definition.function):
pattern7a: function(a) {}

Name: pattern7b
Parameters: (b)
Doc:
// Pattern 7b comment.
Definition (definition.function):
pattern7b: (b) => {}

Name: pattern8
Reference (reference.call):
pattern8(a)

Name: member
Reference (reference.call):
pattern9.member(a)

Name: Pattern10
Reference (reference.class):
new Pattern10(a)

Name: Pattern11
Doc:
// Comment
Definition (definition.class):
const Pattern11 = class Ignore {};

Name: Pattern12
Doc:
// Comment
Definition (definition.class):
export var Pattern12 = class Ignore {};
Metadata:
    is_export = true
